src/lxc/log.c

```diff
@@ -170,10 +170,13 @@ static int log_append_logfile(const struct lxc_log_appender *appender,
 		     event->locinfo->file, event->locinfo->func,
 		     event->locinfo->line);
 
-	n += vsnprintf(buffer + n, sizeof(buffer) - n, event->fmt,
-		       *event->vap);
+	if (n < 0)
+		return n;
 
-	if (n >= sizeof(buffer) - 1) {
+	if (n < sizeof(buffer) - 1)
```

Comment:
That doesn't work. Actually this patch resolves the condition "n >= sizeof(buffer) - 1". If this condition is met, vsnprintf() should not be called at all. In this situation, the first parameter of vsnprintf() will point to the address ">= buffer + sizeof(buffer) - 1", and the second parameter will >=1, allowing vsnprintf() to overwrite any address beyond the address of buffer if n is big enough. This is a typical security vulnerability, allowing carefully constructing an attack payload to overwrite any contents, e.g, the returned address of current function log_append_logfile().

Fortunately, the known applications leveraging log_append_logfile() seem to be not related to escalation of privileges (e.g, set-uid-root), so this issue is not a real CVE bug.